packages/web-components/src/menu-item/menu-item.ts

```diff
@@ -125,48 +88,27 @@ export class MenuItem extends FASTElement {
   @observable
   public slottedSubmenu!: HTMLElement[];
 
-  /**
-   * @internal
-   */
-  public get hasSubmenu(): boolean {
-    return !!this.submenu;
-  }
-
   /**
    * Sets the submenu and updates its position.
    *
    * @internal
    */
   protected slottedSubmenuChanged(prev: HTMLElement[] | undefined, next: HTMLElement[]) {
+    this.submenu?.removeEventListener('toggle', this.toggleHandler);
+
     if (next.length) {
       this.submenu = next[0];
-      this.updateSubmenu();
+      this.submenu.setAttribute('popover', '');
```

Comment:
nit: use `toggleAttribute` for boolean attributes:

```suggestion
      this.submenu.toggleAttribute('popover', true);
```